src/sentry/spans/consumers/process_segments/enrichment.py

```diff
@@ -108,6 +113,8 @@ def _data(self, span: SegmentSpan) -> dict[str, Any]:
             for key, value in shared_tags.items():
                 if ret.get(key) is None:
                     ret[key] = value
+            if self._has_gen_ai_spans and span is self._segment_span:
+                ret["hasGenAISpans"] = True
```

Comment:
Suggesting

```suggestion
                ret["sentry._internal.has_gen_ai_spans"] = True
```

to hide it from the uesr and documenting it in https://github.com/getsentry/sentry-conventions/.